dataikuapi/dss/ml.py

```diff
@@ -308,50 +332,679 @@ def set_metric(self, metric=None, custom_metric=None, custom_metric_greater_is_b
+                self[hyperparameter_name]["range"]["max"] = range_max
+            if nb_values is not None:
+                self[hyperparameter_name]["range"]["nbValues"] = nb_values
+
+        if mode_as_range:
+            self._set_hyperparameter_mode_as_range(hyperparameter_name)
+
+    def _set_categorical_values(self, hyperparameter_name, values=None):
+        self._check_hyperparameter_name(hyperparameter_name)
+        if values is None:
+            warnings.warn("Categorical hyperparameter \"{}\" not modified".format(hyperparameter_name))
+        else:
+            assert isinstance(values, dict), "Invalid values input type for hyperparameter " \
+                                             "\"{}\": ".format(hyperparameter_name) + \
+                                             "must be a dictionary"
+            admissible_values = list(self[hyperparameter_name]["values"].keys())
+            for key in values.keys():
+                assert key in admissible_values, "Unknown categorical value \"" + key + "\". Expected a member of " + str(admissible_values)
+                value_error_message = "Invalid input value for hyperparameter \"{}\", category \"{}\": ".format(hyperparameter_name, key)
+                value_error_message += "expected a {\"enabled\": bool} dictionary"
+                val = values[key]
+                assert isinstance(val, dict), value_error_message
+                assert list(val.keys()) == ["enabled"], value_error_message
+                assert list(type(v) for v in val.values()) == [bool], value_error_message
+            for key in values.keys():
+                self[hyperparameter_name]["values"][key] = values[key]
+
+    def _get_strategy(self):
+        return self._hyperparameter_search_params["strategy"]
+
+    def _get_hyperparameter_mode(self, hyperparameter_name):
+        strategy = self._get_strategy()
+        if strategy == "GRID":
+            return self[hyperparameter_name]["gridMode"]
+        else:
+            return self[hyperparameter_name]["randomMode"]
+
+    def _check_hyperparameter_name(self, hyperparameter_name):
+        assert is_basestring(hyperparameter_name), "Invalid type for hyperparameter name: expecting a string"
+        hyperparameter_names = self._get_all_hyperparameter_names()
+        if hyperparameter_name not in hyperparameter_names:
+            message = "Unknown hyperparameter name: \"{}\". ".format(hyperparameter_name)
+            message += "Expected a member of " + str(hyperparameter_names)
+            raise ValueError(message)
+
+    def _check_hyperparameter_searchable(self, hyperparameter_name):
+        assert isinstance(self[hyperparameter_name], dict), "Hyperparameter \"{}\" cannot be searched".format(hyperparameter_name)
+
+    def _check_hyperparameter_numerical(self, hyperparameter_name):
+        assert "range" in self[hyperparameter_name], "Hyperparameter \"{}\" is not numerical or cannot be searched".format(hyperparameter_name)
```

Comment:
```suggestion
        assert "range" in self[hyperparameter_name], "Hyperparameter \"{}\" is not numerical".format(hyperparameter_name)
```

Always called after `_check_hyperparameter_searchable` so we now that it is searchable